### DISCUSSION TITLE
Library include path

### DISCUSSION THREAD

**Discussion by jackhumphries** [bazelbuild/rules_foreign_cc#708] [Q&A]
Hello,

In my project, I am using the `make` rule to compile an external static library with a Makefile. I want to #include the library's header files in some of my project's source files. My project is compiled with the `cc_binary` rule.

In the [make_simple](https://github.com/bazelbuild/rules_foreign_cc/blob/main/examples/make_simple) example, test_lib.cpp includes the library's header (`liba.h`) with the line `#include "liba.h"`.

In my project, I would like the include path to be prefixed by the library's name. For example, the library's name in the `make_simple` example is `make_lib`, so I would like the include path to be `#include "make_lib/liba.h"`. This is helpful for avoiding confusion and issues when a header file in my project has the same name as a header file in the library.

I've read through the documentation and tried a few things but haven't managed to get this working. Does anyone know how to do this?

Thanks.

**Comment by UebelAndre** [comment]
Could you maybe have the rule install the header files in `$INSTALLDIR` to `${INSTALLDIR}/${target_name}/liba.h`? This way when the include path is added, it's added with the library name in it?

**Reply by jackhumphries** [reply]
Thanks for your help. I first tried setting `out_include_dir` to the target name, but as expected, this did not let me update the include paths accordingly.

I ended up adding a mkdir and a copy to the postfix_script entry like so:
```
mkdir $INSTALLDIR/include/make_lib;
cp $EXT_BUILD_ROOT/external/make_lib/.../*.h $INSTALLDIR/include/make_lib
```

The include paths now work as I like. Here is an example: https://github.com/jackhumphries/bazel-ebpf. Take a look at the `main` target, which depends on `libbpf` in linux.BUILD.

Is this the right way to resolve the issue?

**Reply by shixiong-jiang** [reply]
Hello @jackhumphries, did your workaround work? Did you find any better option? 
I am facing a same issue. Thx.
